architecture/2024-08-resource-types-registration.md

```diff
@@ -0,0 +1,641 @@
+
+| Resource Type      | An API entity that provides CRUDL operations. Resource Types are implemented by a resource provider and belong to a resource provider namespace. Resource types are a new resource type introduced by this proposal. This proposal introduces an API for registering and unregistering resource resource types.                                                                                                                                     |
+| OpenAPI/Schema     | A format for documenting HTTP-based APIs. In this document we're primarily concerned with the parts of OpenAPI pertaining to request/response bodies. |
+| Location/Region | An instance of a loosely-connected deployment of UCP. Regions of a UCP can collaborate to serve route requests between each other. A multi-regional deployment of UCP provides fault-isolation. In a multi-regional deployment of UCP, each region is responsible for processing requests that correspond to its region based on the `location` field. |
+
+## Objectives
+
+> **Issue Reference:** https://github.com/radius-project/radius/issues/6688
+
+### Goals
+
+- Support granular registration of resource providers and resource types (add/change/remove) by users.
+- Support efficient querying of resource type data for any API consumer.
+- Register resource types for existing Radius functionality like Deployment Engine and `Applications.*` resource types.
+- Support the future of UCP as a multi-regional control-plane by avoiding breaking changes in the future.
+  - For now we ensure the `location` field and concept is surfaced in the API by assuming that multi-regional deployments are possible. In practice only a single region will be present.
+
+### Non goals
+
+- (out of scope): Complete the user-defined type implementation. This design document is scoped to registration of resource types.
+- (out of scope): Tooling for resource type registration (CLI, Bicep). This work is tracked by another design document.
+- (non-goal): Provider non-user-facing mechanisms for resource type registration.
+- (out of scope): A complete multi-locational architecture for UCP. 
+
+### User scenarios (optional)
+
+#### Defining new types
+
+Suzy is a platform engineer at Contoso working on the internal developer platform. She defines a resource type for an internal service `ContosoBus` named `Contoso.Platform/contosoBuses`. 
+
+She can use tools like Bicep, GitOps, or the `rad` CLI to register the new resource types in any tenant of Radius.
+
+*Note: the exact set of tooling implementations are out of scope for this document.*
+
+#### Removing default resource types
+
+Suzy is a platform engineer at Contoso working on the internal developer platform. She wants to avoid confusion about the set of resource types supported by the platform.
```

Comment:
Recommend clarification of how she wants to go about with avoiding confusion, i.e. removing registered types that are not applicable to her users.

Reply:
The wording is intentional, and it's consistent with the scenario doc. We take the position in that doc that platform engineers are the curators based on the technologies supported by the platform team.